google-cloud-spanner/src/test/java/com/google/cloud/spanner/IntegrationTestEnv.java

```diff
@@ -162,6 +168,35 @@ private void initializeInstance(InstanceId instanceId) {
     logger.log(Level.INFO, "Created test instance: {0}", createdInstance.getId());
   }
 
+  private void cleanUpOldDatabases(InstanceId instanceId) {
+    int OLD_DB_THRESHOLD_SECS = 24 * 60 * 60;
+    Timestamp currentTimestamp = Timestamp.now();
+    int numDropped = 0;
+
+    Page<Database> page = databaseAdminClient.listDatabases(instanceId.getInstance());
```

Comment:
I think it would be good to filter on the name of the databases as well, so we only automatically delete databases that has the same naming pattern as is generated by the automated tests. If someone for whatever reason manually creates a test database on this instance, it would now also be deleted after 24 hours. Are we sure we want to do that?

Reply:
That makes sense. But filter is not directly supported on listDatabases RPC, so have made a change to match regex after pulling in list of all the databases in the instance.

Reply:
SGTM